frontend-react/src/components/Table/Table.tsx

```diff
@@ -492,7 +551,13 @@ const Table = ({
                         <TableHeaders />
                     </thead>
                     <tbody className="font-mono-2xs">
-                        <TableRows />
+                        <TableRows
+                            rows={memoizedRows}
```

Comment:
yeah, the primary reason was testing. I agree that I could have written the tests that I wrote by rendering an entire table component with tableRows rendered as children, and we still can, even with this refactored to take props, if you'd prefer. 

Without getting too into the weeds in a comment, here's a quick and non prioritized take on why I took the approach I did.

* speed of testing - this way we don't have to render the entire table component for each test. Probably a very negligible gain
* ease of testing - we don't need to wrap the component under test in any particular parents, as we would for table potentially
* speed of render - with TableRows defined within Table, the component is recalculated on each render. Maybe a negligible gain again
* reusability / modularity - with this split out we could use it again. I agree that that doesn't seem likely now, but maybe we end up having multiple types of tables that are different but can all treat rows the same way. In this case we've got a component ready to go. It can also live in its own file now, which will be good because this one is getting pretty unwieldy already.

Reply:
happy to talk this through more, as this gets at the sort of testing philosophy related questions I brought up yesterday

Reply:
I think the confusion came from only the rows and data being migrated and the rest still inside `Table`. I like the split out for maintenance. It's definitely becoming a big file haha. Maybe break those two components out into their own files, and we can break out the rest if/as needed.